### ISSUE TITLE
infer dataset problem

### ISSUE THREAD

**Issue by nlpCSir** [yangheng95/PyABSA#97]
在我使用checkpoint进行测试的时候，我使用了在训练时相同的文件夹名作为target_file,并且在文件夹中建立了.train.txt.inference的文件，可是出现了如下错误。
![1635161741(1)](https://user-images.githubusercontent.com/92669083/138688152-936acd8b-2d24-4025-b333-4f64a52e34a7.png)

**Comment by yangheng95** [comment]
请把你的测试集所在目录结构发给我看一下

**Comment by nlpCSir** [comment]
还有如图所示的这种情况，明明yuan'shu'ju'zhong原数据中只有一个赠品被标注出，可是在进行预测时对两个赠品都进行了预测，我不知道是您故意这么做还是这是一个bug
![1635163804(1)](https://user-images.githubusercontent.com/92669083/138692612-52b79dd1-dd0e-4342-8a85-6cbf9f56b75d.png)

**Comment by nlpCSir** [comment]
![1635163905(1)](https://user-images.githubusercontent.com/92669083/138692684-fae78e7e-7d33-43d8-bfd9-daeee59ed0d1.png)

**Comment by yangheng95** [comment]
> 还有如图所示的这种情况，明明yuan'shu'ju'zhong原数据中只有一个赠品被标注出，可是在进行预测时对两个赠品都进行了预测，我不知道是您故意这么做还是这是一个bug ![1635163804(1)](https://user-images.githubusercontent.com/92669083/138692612-52b79dd1-dd0e-4342-8a85-6cbf9f56b75d.png)

预测已经可以运行了吗?
这是一个bug，当文本中出现多个aspect一样的词语会触发，不过由于工作重心变化，我不会马上修复，如果有人能够PR修复我会非常感激

**Comment by nlpCSir** [comment]
我采取了数组循环的方式进行预测，而使用batch_infer时仍然会出现我前面提到的问题

**Comment by yangheng95** [comment]
请尝试更新findfile, 
`pip install -U findfile`

**Comment by nlpCSir** [comment]
好的，我会尝试，感谢您的回答。前面提到的那个bug，会影响预测，在我进行zh诸如如下图片等数据进行预测时，我发现您的程序会自动对我并没有标注出的数据进行标注，并且出现error警告，特来告知您，
![1635164087(1)](https://user-images.githubusercontent.com/92669083/138693482-937ae540-a800-42e8-8687-4fb3e76410ba.png)

**Comment by yangheng95** [comment]
如果不奏效请将你的预测脚本代码发给我看看

**Comment by nlpCSir** [comment]
好的，我会在尝试后给您答复，目前我还在考虑如何避免那个bug

**Comment by yangheng95** [comment]
fork之后修改infer set的读取，将aspect替换为$+"aspect+"+$等特殊形式。在预测过程中定位修饰后的aspect即可。

**Comment by nlpCSir** [comment]
`
from pyabsa.functional import Trainer
from pyabsa.functional import APCConfigManager
from pyabsa.functional import ABSADatasetList
from pyabsa.functional import APCModelList
from pyabsa.functional import APCCheckpointManager
from pyabsa import  available_checkpoints

config = APCConfigManager.get_apc_config_chinese()
config.dropout = 0.5
config.l2reg = 0.0001
config.model = APCModelList.FAST_LCF_BERT
config.spacy_model = 'zh_core_web_sm'
config.max_seq_len=512
config.polarities_dim=4
config.batch_size=8
config.pretrained_bert="D:/chenchao/bert-base-chinese/"
config.log_step=10000
config.evaluate_begin=0
chinese_sets ="D:/chenchao/testPYABSA/integrated_datasets/apc_datasets/Chinese/Chips"

sent_classifier=APCCheckpointManager.get_sentiment_classifier(checkpoint="fast_lcf_bert_acc_80.16_f1_68.38",auto_device="cuda:0")

results = sent_classifier.batch_infer(target_file=chinese_sets,
                                      print_result=True,
                                      save_result=True,
                                      ignore_error=True,
                                      )
`
这是我的预测代码，我尝试更新findfile至最新版本，但是仍然没有解决问题，报错如下
![1635165268(1)](https://user-images.githubusercontent.com/92669083/138695960-dd98dc09-12d9-4250-9701-82f5576d1781.png)

**Comment by yangheng95** [comment]
只有测试集生成的inference set才会被识别。你用训练集生成了inference set。测试模型时不可使用训练数据，会严重过拟合

**Comment by nlpCSir** [comment]
这里并非是训练数据，我只是模仿您的仓库中的
![image](https://user-images.githubusercontent.com/92669083/138696856-d3323dfc-adc3-4838-b249-49a4d10ea639.png)
数据

**Comment by yangheng95** [comment]
请将train.txt.inference 改为test.txt.inference，或者直接去掉train字样。

**Comment by nlpCSir** [comment]
好的，确实生效了，十分感谢。之前我也尝试过，可能因为拼写错误没有成功。

**Comment by nlpCSir** [comment]
关于infer后的文件的解码，您是否有相关demo呢？

**Comment by yangheng95** [comment]
结果的保存是一个实验性功能，暂未设计读取demo
